prover/src/lib.rs

```diff
@@ -93,7 +98,23 @@ where
                 stack_outputs.clone(),
             );
             #[cfg(all(feature = "metal", target_arch = "aarch64", target_os = "macos"))]
-            let prover = gpu::MetalRpoExecutionProver(prover);
+            let prover = gpu::MetalExecutionProver::<RpoRandomCoin, Rpo256, RpoDigest>::new(
+                prover,
+                HashFn::Rpo256,
+            );
+            prover.prove(trace)
+        }
+        HashFunction::Rpx256 => {
+            let prover = ExecutionProver::<Rpx256, RpxRandomCoin>::new(
+                options,
+                stack_inputs,
+                stack_outputs.clone(),
+            );
+            #[cfg(all(feature = "metal", target_arch = "aarch64", target_os = "macos"))]
+            let prover = gpu::MetalExecutionProver::<RpxRandomCoin, Rpx256, RpxDigest>::new(
+                prover,
+                HashFn::Rpx256,
+            );
```

Comment:
nit: could we follow the same sequence of type parameters as for execution prover - i.e., `<Rpx256, RpxDigest, RpxRandomCoin>`?

Also, do we actually need the type parameter for digest? Or could we infer it from the hasher. That is, could we change `MetalExecutionProver` to accept only two type parameters as `<Rpx256, RpxRandomCoin>` (similar to how regular `ExecutionProver` works.